### PR TITLE
Docs: Fix the API reference

### PR DESCRIPTION
**Pull request by nik9000**
Fixes some outbound links in the API reference. These are caused by bad
links in Elasticsearch's json spec files. We've filed and issue with
Elasticsearch here:
https://github.com/elastic/elasticsearch/issues/40596

but for now this works around the bad links.

### CODE REVIEW

File: scripts/utils/generateDocs.js [hunk 1]
```diff
@@ -140,13 +143,29 @@ function generateApiDoc (spec) {
...
+  "migration.deprecations": "{ref}/migration-api-deprecation.html",
+  "monitoring.bulk": "{ref}/es-monitoring.html",
+}
 // Fixes bad urls in the JSON spec
-function fixLink (str) {
+function fixLink (name, str) {
+  var override = LINK_OVERRIDES[name]
+  if (override) return override
   if (!str) return ''
-  if (str.includes('/5.x/')) {
```
Comment (nik9000) [inline scripts/utils/generateDocs.js]: Elasticsearch has all kinds of wrong branches linked and this new replacement forces them all to the "current" branch.

File: scripts/utils/generateDocs.js [hunk 2]
```diff
@@ -140,13 +143,29 @@ function generateApiDoc (spec) {
...
+const LINK_OVERRIDES = {
+  "license.delete": "{ref}/delete-license.html",
+  "license.get": "{ref}/get-license.html",
+  "license.get_basic_status": "{ref}/get-basic-status.html",
+  "license.get_trial_status": "{ref}/get-trial-status.html",
+  "license.post": "{ref}/update-license.html",
+  "license.post_start_basic": "{ref}/start-basic.html",
+  "license.post_start_trial": "{ref}/start-trial.html",
+  "migration.deprecations": "{ref}/migration-api-deprecation.html",
+  "monitoring.bulk": "{ref}/es-monitoring.html",
```
Comment (nik9000) [inline scripts/utils/generateDocs.js]: Some of these can't be fixed by regex and others would be complex to fix via regex so I do it this way here.

**Comment by delvedor** [comment]
I've backported the fix to `7.x`, `6.x`, and `5.x`, I've regenerated the reference as well.

**Comment by nik9000** [comment]
Thanks! This seems to have fixed all of the broken links in 7.x and master. There are a few in 6.x and 5.x:

```
INFO:build_docs:Bad cross-document links:
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/client/javascript-api/5.x/api-reference.html:
INFO:build_docs:   - en/elasticsearch/plugins/5.x/ingest.html
INFO:build_docs:   - en/elasticsearch/reference/master/search-field-stats.html
INFO:build_docs:   - en/elasticsearch/reference/master/search-percolate.html
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/elasticsearch/client/javascript-api/6.x/api-reference.html:
INFO:build_docs:   - en/elasticsearch/reference/6.x/secure-settings.html
INFO:build_docs:   - en/elasticsearch/reference/master/watcher-api-restart.html
INFO:build_docs:   - en/migration/current/migration-api-deprecation.html
INFO:build_docs:   - en/monitoring/current/appendix-api-bulk.html
```

I can look at those in a bit.

**Pull request merged**